tests/unit/test_event_service_create_event.py

```
@@ -132,3 +138,33 @@ async def test_create_event_subject_not_found_raises(
     assert exc_info.value.details.get("resource_type") == "subject"
     assert exc_info.value.details.get("resource_id") == "sub1"
     event_repo.create_event.assert_not_awaited()
+
+
+async def test_create_event_idempotency_returns_existing(
+    event_service_mocks,
+) -> None:
+    """When external_id is set and event already exists, return it without creating."""
+    svc, event_repo, subject_repo = event_service_mocks
```

Ruff (0.15.5)

[warning] Unpacked variable `subject_repo` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)